Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1

```diff
@@ -61,7 +61,8 @@ $personalAccessToken = $vssEndpoint.Authorization.Parameters.AccessToken
 
 if ( [string]::IsNullOrEmpty($personalAccessToken))
 {
-  throw (Get-LocalizedString -Key "Unable to generate Personal Access Token for the user. Contact Project Collection Administrator")
+    Write-Host "##vso[task.logissue type=error;code=001002;]"
+    throw (Get-LocalizedString -Key "Unable to generate Personal Access Token for the user. Contact Project Collection Administrator")
```

Comment:
not your code ... but some throws are using Get-LocalizedString, while others are not. Assuming that its required for getting localized strings, we can use it at other places as well.